db/Pos/DB/Error.hs

```diff
@@ -9,13 +9,15 @@ import           Formatting          (bprint, stext, (%))
 import           Universum
 
 data DBError =
-    -- | Structure of DB is malformed (e. g. data is inconsistent,
-    -- something is missing, etc.)
-    DBMalformed !Text
+      -- | Structure of DB is malformed (e. g. data is inconsistent,
+      -- something is missing, etc.)
+      DBMalformed !Text
+    | DBUnexpectedVersionTag !Text
```

Comment:
It's better to have 2 `Word8` here (expected and real) and take care of formatting in `Buildable` instance, rather than having formatted `Text`. For example, if later someone wants to write tests and check whether this exception is thrown with proper values, they will be able to check numbers directly.